process-document/src/test/kotlin/com/ritense/processdocument/service/CorrelationServiceIntTest.kt

```diff
@@ -0,0 +1,219 @@
+package com.ritense.processdocument.service
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.ritense.document.domain.Document
+import com.ritense.document.domain.impl.JsonSchemaDocumentId
+import com.ritense.document.domain.impl.request.NewDocumentRequest
+import com.ritense.document.service.DocumentService
+import com.ritense.processdocument.BaseIntegrationTest
+import com.ritense.processdocument.repository.ProcessDocumentInstanceRepository
+import com.ritense.valtimo.service.CamundaProcessService
+import org.camunda.bpm.engine.RuntimeService
+import org.camunda.bpm.engine.TaskService
+import org.camunda.bpm.engine.task.Task
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.transaction.annotation.Transactional
+import java.util.function.Consumer
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+@Transactional
+class CorrelationServiceIntTest: BaseIntegrationTest() {
```

Comment:
I don't think there's enough scenarios covered here. I would expect each method to have at least two tests, maybe more. For example, when writing a test for `sendMessageToAll(message: String, businessKey: String?, variables: Map<String,Any>?)`, I would write the following tests:

1. 1 message catch event, matching businesskey
2. 0 message catch events
3. 3 message catch events with only 2 having a matching business key 
4. 1 message catch events, matching businesskey, with variables